File: android2_lesson04/src/java/java/com/example/ama/android2_lesson04/ui/viewer/PVJobServiceFragment.java
package com.example.ama.android2_lesson04.ui.viewer;

import android.os.Bundle;
import android.support.annotation.Nullable;

import com.example.ama.android2_lesson04.background.BackgroundConstants;
import com.example.ama.android2_lesson04.background.service.LoadPictureJobService;
import com.example.ama.android2_lesson04.ui.viewer.base.BaseReceiverFragment;
import com.example.ama.android2_lesson04.utils.ResourcesUtils;
import com.firebase.jobdispatcher.Constraint;
import com.firebase.jobdispatcher.FirebaseJobDispatcher;
import com.firebase.jobdispatcher.GooglePlayDriver;
import com.firebase.jobdispatcher.Job;
import com.firebase.jobdispatcher.Lifetime;
import com.firebase.jobdispatcher.Trigger;

public class PVJobServiceFragment extends BaseReceiverFragment {
    private FirebaseJobDispatcher dispatcher;

    public static PVJobServiceFragment newInstance() {
        return new PVJobServiceFragment();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        setRetainInstance(true);
        dispatcher = new FirebaseJobDispatcher(new GooglePlayDriver(getContext()));
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onLoadImagesClick() {
        scheduleLoadingImagesJob();
    }

    private void scheduleLoadingImagesJob() {
        Bundle bundle = new Bundle();
        bundle.putStringArray(BackgroundConstants.EXTRA_KEY, ResourcesUtils.getUrlsArray());
        Job job = dispatcher.newJobBuilder()
                .setService(LoadPictureJobService.class)
                .setExtras(bundle)
                .setTrigger(Trigger.NOW)
                .setLifetime(Lifetime.UNTIL_NEXT_BOOT)
                .setReplaceCurrent(false)
                .setConstraints(Constraint.ON_ANY_NETWORK)
                .setTag(LoadPictureJobService.class.getSimpleName())
                .build();
        dispatcher.schedule(job);
    }

    @Override
    public void onDestroy() {
        dispatcher.cancel(LoadPictureJobService.class.getSimpleName());
        super.onDestroy();
    }
}

File: android2_lesson04/src/kotlin/java/com/example/ama/android2_lesson04/ui/viewer/PVJobServiceFragment.kt
package com.example.ama.android2_lesson04.ui.viewer

import android.os.Bundle
import com.example.ama.android2_lesson04.R
import com.example.ama.android2_lesson04.background.service.LoadPictureJobService
import com.example.ama.android2_lesson04.background.utils.EXTRA_KEY
import com.example.ama.android2_lesson04.ui.viewer.base.BaseReceiverFragment
import com.firebase.jobdispatcher.*

class PVJobServiceFragment : BaseReceiverFragment() {
    private lateinit var dispatcher: FirebaseJobDispatcher

    companion object {
        fun newInstance() = PVJobServiceFragment()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        dispatcher = FirebaseJobDispatcher(GooglePlayDriver(context))
        retainInstance = true
        super.onCreate(savedInstanceState)
    }

    override fun onDestroy() {
        dispatcher.cancel(LoadPictureJobService::class.java.simpleName)
        super.onDestroy()
    }

    override fun onLoadImagesClick() {
        scheduleLoadingImagesJob()
    }

    private fun scheduleLoadingImagesJob() {
        val bundle = Bundle()
        bundle.putStringArray(EXTRA_KEY, resources.getStringArray(R.array.image_urls))
        val job = dispatcher.newJobBuilder()
                .setService(LoadPictureJobService::class.java)
                .setExtras(bundle)
                .setLifetime(Lifetime.UNTIL_NEXT_BOOT)
                .setTrigger(Trigger.NOW)
                .setTag(LoadPictureJobService::class.java.simpleName)
                .setConstraints(Constraint.ON_ANY_NETWORK)
                .build()
        dispatcher.schedule(job)
    }
}